ui/src/app/event-sources/components/event-source-details/event-source-details.tsx

```diff
@@ -0,0 +1,159 @@
+    const [selectedNode, setSelectedNode] = useState<string>(queryParams.get('selectedNode'));
+
+    useEffect(
+        () =>
+            history.push(
+                historyUrl('event-sources/{namespace}/{name}', {
+                    namespace,
+                    name,
+                    tab,
+                    selectedNode
+                })
+            ),
+        [namespace, name, tab, selectedNode]
+    );
+
+    const [edited, setEdited] = useState(false);
+    const [error, setError] = useState<Error>();
+    const [eventSource, setEventSource] = useState<EventSource>();
+
+    const selected = (() => {
+        if (!selectedNode) {
+            return;
+        }
+        const x = ID.split(selectedNode);
+        const value = eventSource;
+        return {value, ...x};
+    })();
+
+    useEffect(() => {
+        services.eventSource
+            .get(name, namespace)
+            .then(setEventSource)
+            .then(() => setEdited(false)) // set back to false
+            .then(() => setError(null))
+            .catch(setError);
+    }, [name, namespace]);
+
+    useEffect(() => setEdited(true), [eventSource]);
+
+    return (
+        <Page
+            title='Event Source Details'
+            toolbar={{
+                breadcrumbs: [
+                    {title: 'Event Source', path: uiUrl('event-sources')},
+                    {title: namespace, path: uiUrl('event-sources/' + namespace)},
+                    {title: name, path: uiUrl('event-sources/' + namespace + '/' + name)}
+                ],
+                actionMenu: {
+                    items: [
+                        {
+                            title: 'Update',
+                            iconClassName: 'fa fa-save',
+                            disabled: !edited,
+                            action: () =>
+                                services.eventSource
+                                    .update(eventSource, name, namespace)
+                                    .then(setEventSource)
+                                    .then(() =>
+                                        notifications.show({
+                                            content: 'Updated',
+                                            type: NotificationType.Success
+                                        })
+                                    )
+                                    .then(() => setEdited(false))
+                                    .then(() => setError(null))
+                                    .catch(setError)
+                        },
+                        {
+                            title: 'Delete',
+                            iconClassName: 'fa fa-trash',
+                            disabled: edited,
+                            action: () => {
+                                popup.confirm('Confirm', 'Are you sure you want to delete this event source?\nThere is no undo.').then(yes => {
```

Comment:
this looks different to other uses - making me thing those other uses are wrong - can you please check them and fix if needed?

Reply:
only the workflow page is using this. I will do it separate PR